src/runtime_src/driver/xclng/xrt/user_gem/perf.cpp

```diff
@@ -961,6 +962,25 @@ namespace xocl {
     return size;
   }
 
+  int XOCLShim::xclGetDebugProfileDeviceInfo(xclDebugProfileDeviceInfo& info) {
+    auto dev = pcidev::get_dev(mBoardNumber);
+    uint16_t user_instance = dev->user->instance;
+    uint16_t mgmt_instance = dev->mgmt->instance;
+    uint16_t nifd_instance = 0;
+    std::string device_name = std::string(DRIVER_NAME_ROOT) + std::string(DEVICE_PREFIX) + std::to_string(user_instance);
+    std::string nifd_name = std::string(DRIVER_NAME_ROOT) + std::string(NIFD_PREFIX) + std::to_string(nifd_instance);
+    std::string sysfs_name = SYSFS_NAME_ROOT + dev->user->sysfs_name;
+    info.device_type = DeviceType::XBB;
+    info.device_index = mBoardNumber;
+    info.user_instance = user_instance;
+    info.mgmt_instance = mgmt_instance;
+    info.nifd_instance = nifd_instance;
+    strncpy(info.device_name, device_name.c_str(), MAX_NAME_LEN);
+    strncpy(info.sysfs_name, sysfs_name.c_str(), MAX_NAME_LEN);
+    strncpy(info.nifd_name, nifd_name.c_str(), MAX_NAME_LEN);
```

Comment:
Please check the return value of strncpy and possibly manually null terminate.

Reply:
or just blindly put a 0 at the end of the char array after calling strncpy().

Reply:
I just pushed a change to forced all char array to have a null at the end:

```cpp
info.device_name[MAX_NAME_LEN-1] = '\0';
info.sysfs_name[MAX_NAME_LEN-1] = '\0';
info.nifd_name[MAX_NAME_LEN-1] = '\0';
```